### DISCUSSION TITLE
[RISOLTO] Tutte le integrazioni cloud su HA non funzionano!

### DISCUSSION THREAD

**Discussion by chip83** [indomus/forum#2952] [HUB: Home Assistant]
Non so cosa sia successo, ma dalle 17 circa di oggi, tutte le integrazioni cloud su HA hanno smesso di funzionare...
cosa può essere? è tutto installato su un miniPC con debian ed il pc è connesso ad internet in quanto raggiungibile (HA) dall'esterno...
Ho riavviato HA senza risolvere...idee?

**Comment by snakuzzo** [comment]
Devi guardare i log. Come pretendi che uno possa darti una risposta in base a quello che hai detto? Mica sappiamo leggere nella sfera di cristallo?!

**Reply by chip83** [reply]
si scusate, era tarti ed avevo intenzione di integrare con più info stamattina.
Allora, **questo è un estratto del log** dove si ripete sostanzialmente questa cosa:

```
2024-03-04 08:14:36.102 WARNING (MainThread) [custom_components.senec.pysenec_ha] APP-API: NO RES - Could not get data from https://app-gateway.prod.senec.dev/v1/senec/monitor/266017/data/custom?startDate=2018-01-01&endDate=2024-04-01&locale=de_DE&timezone=GMT causing: Cannot connect to host app-gateway.prod.senec.dev:443 ssl:False [Try again]

2024-03-04 08:14:41.110 WARNING (MainThread) [custom_components.senec.pysenec_ha] APP-API: NO RES - Could not get data from https://app-gateway.prod.senec.dev/v1/senec/systems/295037/technical-data causing: Cannot connect to host app-gateway.prod.senec.dev:443 ssl:False [Try again]

2024-03-04 08:14:46.118 WARNING (MainThread) [custom_components.senec.pysenec_ha] APP-API: NO RES - Could not get data from https://app-gateway.prod.senec.dev/v1/senec/systems/295037/technical-data causing: Cannot connect to host app-gateway.prod.senec.dev:443 ssl:False [Try again]

2024-03-04 08:14:46.164 ERROR (MainThread) [homeassistant.components.forecast_solar] Unexpected error fetching forecast_solar data: Error while resolving Forecast.Solar API address
Traceback (most recent call last):
File "/usr/local/lib/python3.12/site-packages/forecast_solar/__init__.py", line 78, in _request
result = await dns.query("api.forecast.solar", "A")
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
aiodns.error.DNSError: (11, 'Could not contact DNS servers')
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 313, in _async_refresh
self.data = await self._async_update_data()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/forecast_solar/coordinator.py", line 67, in _async_update_data
return await self.forecast.estimate()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/forecast_solar/__init__.py", line 156, in estimate
data = await self._request(
^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/forecast_solar/__init__.py", line 80, in _request
raise ForecastSolarConnectionError(
forecast_solar.exceptions.ForecastSolarConnectionError: Error while resolving Forecast.Solar API address

2024-03-04 08:14:51.140 ERROR (MainThread) [homeassistant.components.homeassistant_alerts] Error requesting homeassistant_alerts data: Cannot connect to host alerts.home-assistant.io:443 ssl:default [Try again]

2024-03-04 08:14:51.155 ERROR (MainThread) [metno] Access to https://aa12h6berfevih82i1.api.met.no/weatherapi/locationforecast/2.0/complete returned error 'ClientConnectorError'

2024-03-04 08:14:56.171 ERROR (MainThread) [custom_components.hacs] Could not update template - Error fetching data from HACS: Cannot connect to host data-v2.hacs.xyz:443 ssl:default [Try again]

2024-03-04 08:14:56.171 ERROR (MainThread) [custom_components.hacs] Could not update integration - Error fetching data from HACS: Cannot connect to host data-v2.hacs.xyz:443 ssl:default [Try again]

2024-03-04 08:14:56.172 ERROR (MainThread) [custom_components.hacs] Could not update theme - Error fetching data from HACS: Cannot connect to host data-v2.hacs.xyz:443 ssl:default [Try again]

2024-03-04 08:14:56.172 ERROR (MainThread) [custom_components.hacs] Could not update plugin - Error fetching data from HACS: Cannot connect to host data-v2.hacs.xyz:443 ssl:default [Try again]
```

**sembra** quindi **che HA non riesca a collegarsi ad internet**, ma:

- è accessibile dall'esterno (altra connessione, tramite IP pubblico)
- se mi loggo tramite SSH alla macchina ed eseguo un ping verso un sito qualsiasi, questo va...

Ripensandoci, questo problema l'ho riscontrato ieri sera dopo aver fatto:
- creazione nuovo account su linux debian
- riavvio del miniPC con linux debian
- installazione e configurazione di motionEye con una videocamera

le cose sono avvenute in un certo lasso di tempo, è più probabile che il problema si sia manifestato o in concomitanza con il **riavvio** (molto probabile) o con l'installazione di motionEye.

Inoltre, è normale che su rete mi dica "accesso esterno disabiltiato"? ...e se entro clicco sulla voce del menu mi apra una pagina completamente bianca?
![Screenshot 2024-03-04 094238](https://github.com/indomus/forum/assets/22590434/5ae91283-5a6c-4cce-8562-30826b0c0624)

**Reply by snakuzzo** [reply]
Innanzi tutto...correggi il log incluso nel markdown. Si tratta di un blocco di log, non di una riga. Così come hai scritto è un "bordello".

Tornando al problema, non è che non riesce a collegarsi ad internet...è che hai problemi con la risoluzione DNS

**Reply by chip83** [reply]
ho sistemato il log...
problemi di DNS in che senso?
Se da una sessione SSH pingo uno di quei server ho:

```
deb@miniPC:~$ ping app-gateway.prod.senec.dev
PING app-gateway.prod.senec.dev (20.79.230.201) 56(84) bytes of data.
64 bytes from 20.79.230.201 (20.79.230.201): icmp_seq=1 ttl=110 time=102 ms
64 bytes from 20.79.230.201 (20.79.230.201): icmp_seq=2 ttl=110 time=124 ms
64 bytes from 20.79.230.201 (20.79.230.201): icmp_seq=3 ttl=110 time=52.6 ms
64 bytes from 20.79.230.201 (20.79.230.201): icmp_seq=4 ttl=110 time=67.2 ms
64 bytes from 20.79.230.201 (20.79.230.201): icmp_seq=5 ttl=110 time=90.6 ms
64 bytes from 20.79.230.201 (20.79.230.201): icmp_seq=6 ttl=110 time=113 ms
```

quindi i nomi di dominio li risolve...
ad intuito, potrebbe essere qualcosa all'interno del container docker su cui è instanziato HA?
Come posso verificare, non me ne intendo...

**Reply by snakuzzo** [reply]
Tu questo lo stai facendo dall'host.
Prova a farlo dal container di core...inoltre il test di connettività non si fa con il ping, perché l'ICMP non è detto sia consentito sul target, al massimo ti serve solo per capire se e come viene risolto.
Quindi...

```
docker exec -it <nome_container_ha> bash
```

e poi...di seguito un paio di comandi ed un esempio di quello che dovrebbe essere l'output corretto

```
/config # nslookup hacs.xyz
Server:		1.1.1.2
Address:	1.1.1.2#53

Non-authoritative answer:
Name:	hacs.xyz
Address: 104.21.39.42
Name:	hacs.xyz
Address: 172.67.143.44
Name:	hacs.xyz
Address: 2606:4700:3031::ac43:8f2c
Name:	hacs.xyz
Address: 2606:4700:3034::6815:272a

/config # nc -zvw2 hacs.xyz 443
hacs.xyz (172.67.143.44:443) open
```

**Reply by chip83** [reply]
ok, non risolve i nomi di dominio...

come risolvo? :D

```
miniPC:/config# nslookup hacs.xyz
;; communications error to ::1#53: connection refused
;; communications error to ::1#53: connection refused
;; communications error to ::1#53: connection refused
;; communications error to 127.0.0.1#53: connection refused
;; no servers could be reached
```
se faccio un ping lo raggiunge, quindi presumo che il problema sia solo il DNS...

```
miniPC:/config# ping 172.67.143.44:443
PING 172.67.143.44:443 (172.67.143.44): 56 data bytes
64 bytes from 172.67.143.44: seq=0 ttl=55 time=63.138 ms
64 bytes from 172.67.143.44: seq=1 ttl=55 time=123.012 ms
64 bytes from 172.67.143.44: seq=2 ttl=55 time=52.533 ms
64 bytes from 172.67.143.44: seq=3 ttl=55 time=67.464 ms
64 bytes from 172.67.143.44: seq=4 ttl=55 time=91.429 ms
64 bytes from 172.67.143.44: seq=5 ttl=55 time=117.973 ms
```

**Reply by snakuzzo** [reply]
Sì, il problema, come ti dicevo prima, è solo la risoluzione DNS.
il problema è che il container ha come DNS server la loopback ipv6 (`::1`) che però sulla porta 53 (quella di default dei DNS server) non è attivo.
Non ti assicuro nulla (a te la responsabilità di quello che fai), ma forse la soluzione veloce, se non ti interessa avere attivo ipv6, è disabilitarlo proprio sull'host...

```
sudo nano /etc/sysctl.conf
```

e in fondo aggiungi

```
# Disabling the IPv6
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1
```

poi, per far prendere la nuova configurazione...

```
sudo sysctl -p
```

Poi dovresti provare a restartare tutti i container (oppure reboot dell'host)

Oppure, puoi provare a vedere se funziona disabilitando lo stub listener...

Es.

```
sudo nano /etc/systemd/resolved.conf
```

e poi restart del servizio...

```
sudo systemctl systemd-resolved.service restart
```

e restart dei container (oppure reboot dell'host)

**Reply by chip83** [reply]
ho disabilitato l'ipv6 e restartato tutti i container, ma nulla, il problema persiste...
```
CONTAINER ID   IMAGE                                          COMMAND                  CREATED        STATUS         PORTS                                                                                         NAMES
32dc98fb2303   ccrisan/motioneye:master-amd64                 "/bin/sh -c 'test -e…"   21 hours ago   Up 5 minutes                                                                                                 motioneye
3416b925d081   koenkk/zigbee2mqtt                             "docker-entrypoint.s…"   46 hours ago   Up 3 minutes                                                                                                 zigbee2mqtt
f617cdaf208e   eclipse-mosquitto                              "/docker-entrypoint.…"   3 days ago     Up 4 minutes   0.0.0.0:1883->1883/tcp, :::1883->1883/tcp                                                     mosquitto
078c67f21c7b   caddy:latest                                   "caddy run --config …"   3 days ago     Up 4 minutes   0.0.0.0:80->80/tcp, :::80->80/tcp, 0.0.0.0:443->443/tcp, :::443->443/tcp, 443/udp, 2019/tcp   caddy
33106138e04a   ghcr.io/home-assistant/home-assistant:stable   "/init"                  3 days ago     Up 4 minutes                                                                                                 home-assistant
```
EDIT: un dubbio... ma il file sysctl.conf lo devo modificare fuori dal container no? :D

invece per la seconda soluzione, nel secondo comando, il sistema mi dice che non trova "systenctl"...

**Reply by snakuzzo** [reply]
è `systemctl` (avevo sbagliato a scrivere)

**Reply by snakuzzo** [reply]
Prova a restartare l'host (fai un reboot)

**Reply by chip83** [reply]
adesso provo con systemctl...
L'host l'avevo già restartato 2 o 3 volte :D

controllando vedo che il file /etc/resolv.conf ha questo nameserver:

```
# Generated by NetworkManager
nameserver fd00::3e37:12ff:fe44:5b23
```

aggiungendone dei miei, NetworkManager, me li riscrive...
Penso che il problema sia questo...come posso fare a modificare il file da terminale? non ho accesso fisico alla macchina in questo momento

**Reply by snakuzzo** [reply]
mai modificare quel file a mano. Il file è in link simbolico ad altro file. E comunque dipende da come è configurato il tuo host
Però permettimi di dire... se hai scelto questo tipo di installazione, devi essere in grado di poterlo gestire.

Quindi -> google

**Reply by chip83** [reply]
finalmente risolto! :D
Alla fine ho riavviato il container di HA per l'ennesima volta ed è andato...

Grazie mille, iniziavo a perdere ogni speranza...reinstallare tutto dopo solo 2 giorni di funzionamento proprio non mi andava...
Tieni conto che ho ordinato tutto, installato e configurato nella pausa pranzo, se percaso dovevo rifare tutto mi sarei sparato prima :D

**Comment by mosiser** [comment]
Se può essere utile per chi si imbatte nello stesso inconveniente...

Stesso problema, installazione ex-novo Docker su Debian 13 (seguendo le guide di inDomus).
Sembra si presenti solo dopo un riavvio dell'host, mentre funziona tutto bene ad un riavvio del container ad host già avviato.
In poche parole, il resolv.conf del container (che si popola automaticamente dal resolv.conf dell'host) rimane vuoto quando la macchina viene riavviata, come se venisse "popolato" troppo presto, quando il file di origine non è ancora pronto.

Scappatoie: riavviare i container dopo qualche minuto, rendere il resolv.conf dell'host immutabile (invece di venir ricreato da dhclient ad ogni boot), oppure più semplicemente aggiungere il dns direttamente al docker-compose...

**Reply by snakuzzo** [reply]
Questa cosa è strana...e comunque la scappatoia non può essere una soluzione al problema. Il resolv.conf sarebbe invece cosa buona e giusta farlo creare in automatico.
Potresti sì forzarlo nel docker-compose, ma è una pezza che fa troppo schifo :)

Premesso che non ho chiaro se questa cosa si verifichi solo su Debian 13 (io sono ancora su Debian 12), hai provato a verificare il service di docker com'è configurato nella parte Unit ?

Ti riporto la mia configurazione:

```shell
$ cat /lib/systemd/system/docker.service
[Unit]
Description=Docker Application Container Engine
Documentation=https://docs.docker.com
After=network-online.target nss-lookup.target docker.socket firewalld.service containerd.service time-set.target
Wants=network-online.target containerd.service
Requires=docker.socket
StartLimitBurst=3
StartLimitIntervalSec=60

[Service]
Type=notify
# the default is not to use systemd for cgroups because the delegate issues still
# exists and systemd currently does not support the cgroup feature set required
# for containers run by docker
ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock
ExecReload=/bin/kill -s HUP $MAINPID
TimeoutStartSec=0
RestartSec=2
Restart=always

# Having non-zero Limit*s causes performance problems due to accounting overhead
# in the kernel. We recommend using cgroups to do container-local accounting.
LimitNPROC=infinity
LimitCORE=infinity

# Comment TasksMax if your systemd version does not support it.
# Only systemd 226 and above support this option.
TasksMax=infinity

# set delegate yes so that systemd does not reset the cgroups of docker containers
Delegate=yes

# kill only the docker process, not all processes in the cgroup
KillMode=process
OOMScoreAdjust=-500

[Install]
WantedBy=multi-user.target
```

**Reply by mosiser** [reply]
Certo, sono scappatoie per mettere una pezza se non si riesce a trovare la soluzione... 
E data la mia inesperienza, già tanto che abbia trovato le scappatorie per far funzionare il tutto... 😀

Controllato il docker.service, è una copia perfetta del tuo.

Sembrerebbe essere un problema con qualche anno sulle spalle https://forums.docker.com/t/dns-issue-due-to-race-condition-between-dhcp-and-docker/109641
Magari dipende da dhclient... Tu hai NetworkManager installato?

**Reply by mosiser** [reply]
Confermo, sembra esserci una "race condition" tra l'avvio di Docker e l'inizializzazione della rete.
O il minipc è troppo veloce, o il server DHCP troppo lento...

Se anche passo esplicitamente il DNS in Docker Compose, ho comunque una serie di errori iniziali di mancata connessione in HA, dovuti al fatto che HA si è già avviato quando la rete non era ancora pronta.

Ho provato ad avviare docker con un pelo di ritardo (aggiungendo `ExecStartPre=/bin/sleep 15` al docker.service), e infatti tutto funziona perfettamente, nessun errore di mancata connessione in HA, e DNS correttamente ricopiati dal resolv.conf dell'host al resolv.conf del container.

Stesso risultato l'ho ottenuto sostituendo `allow-hotplug` con `auto` in /etc/network/interfaces, facendo sì che l'interfaccia di rete si attivi subito al boot, senza attendere il rilievo della connessione fisica...
